x-pack/plugins/observability/public/components/threshold/components/alert_details_app_section.tsx

```diff
@@ -208,24 +199,85 @@ export default function AlertDetailsAppSection({
     </>
   ) : null;
 
+  const cpuMetricPrefix = 'system.cpu';
+  const memoryMetricPrefix = 'system.memory';
+  const relatedMetrics = [
+    'system.cpu.user.pct',
+    'system.load.1',
+    'system.memory.actual.used.pct',
+    'system.filesystem.used.pct',
+    'host.network.ingress.bytes',
+    'host.network.egress.bytes',
+  ];
+  const fnList = ['avg', 'sum', 'min', 'max'];
+
+  const isCpuOrMemoryCriterion = (criterion: MetricExpression) =>
+    criterion.metrics?.some(
+      (metric: CustomThresholdExpressionMetric) =>
+        metric.field?.includes(cpuMetricPrefix) || metric.field?.includes(memoryMetricPrefix)
+    );
+
+  const relatedMetricsPerCriteria = () => {
+    const hasCpuOrMemoryCriteria = ruleParams.criteria.some((criterion) =>
+      isCpuOrMemoryCriterion(criterion)
+    );
+
+    const relatedMetricsInDataView = hasCpuOrMemoryCriteria
+      ? dataView?.fields
+          .map((field) => field.name)
+          .filter((fieldName) => relatedMetrics.includes(fieldName))
+      : [];
+
+    const aggType = ruleParams.criteria
+      .find((criterion) => isCpuOrMemoryCriterion(criterion))
+      ?.metrics?.find(
+        (metric) =>
+          metric.field?.includes(cpuMetricPrefix) || metric.field?.includes(memoryMetricPrefix)
+      )?.aggType;
+
+    const metricAggType = fnList.includes(aggType || '') ? aggType : 'avg';
+
+    return { relatedMetricsInDataView, metricAggType };
+  };
+
+  const relatedEventsTimeRange = (): TimeRange => {
+    return {
+      from: moment(alert.start)
+        .subtract(
+          (ruleParams.criteria[0].timeSize ?? 5) * 2,
+          ruleParams.criteria[0].timeUnit ?? 'minutes'
+        )
+        .toISOString(),
+      to: moment(alert.lastUpdated).toISOString(),
```

Comment:
Question: For recovered alerts, is this when the alert was recovered?

Reply:
yes